nilearn/maskers/nifti_labels_masker.py

```diff
@@ -378,15 +377,6 @@ def _post_masking_atlas(self, visualize=False):
 
     def generate_report(self):
         """Generate a report."""
-        if not is_matplotlib_installed():
-            with warnings.catch_warnings():
-                mpl_unavail_msg = (
-                    "Matplotlib is not imported! No reports will be generated."
-                )
-                warnings.filterwarnings("always", message=mpl_unavail_msg)
-                warnings.warn(category=ImportWarning, message=mpl_unavail_msg)
-                return [None]
-
```

Comment:
Could `generate_report` be implemented in the `BaseMasker` to avoid multiple similar implementations of `generate_report` ?  

```
class BaseMasker:

    def _has_plotting_dependencies(self):
        ....

    def generate_report(self):
         if not self._has_plotting_dependencies():
             return None
          
         from nilearn.reporting.html_report import generate_report

        # this function is to be used by maskers, which change self before generating report
         self._prepare_self()

         return generate_report(self)
```

Reply:
maybe but some maskers do a few extra things before calling nilearn.reporting.html_report.generate_report
so for now I wanted to keep it simple and refactor the easiest bit first.